addons/metadata.integral.artists/lib/musicbrainz.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: UTF-8 -*-
+def musicbrainz_artistfind(data):
+    artists = []
+    for item in data.get('artists',[]):
+        artistdata = {}
+        artistdata['artist'] = item['name']
+        artistdata['thumb'] = ''
+        artistdata['genre'] = ''
+        artistdata['born'] = item['life-span'].get('begin', '')
+        if 'type' in item:
+            artistdata['type'] = item['type']
+        if 'gender' in item:
+            artistdata['gender'] = item['gender']
+        if 'disambiguation' in item:
+            artistdata['disambiguation'] = item['disambiguation']
+        artistdata['mbid'] = item['id']
+        if item.get('score',1):
+            artistdata['relevance'] = str(item['score'] / 100.00)
+        artists.append(artistdata)
+    return artists
+
+def musicbrainz_artistdetails(data):
+    artistdata = {}
+    artistdata['artist'] = data['name']
+    artistdata['mbartistid'] = data['id']
+    artistdata['type'] = data['type']
+    artistdata['gender'] = data['gender']
+    artistdata['disambiguation'] = data['disambiguation']
+    if data.get('life-span','') and data.get('type',''):
+        begin = data['life-span'].get('begin', '')
+        end = data['life-span'].get('end', '')
+        if data['type'] in ['Group', 'Orchestra', 'Choir']:
+            artistdata['formed'] = begin
+            artistdata['disbanded'] = end
+        elif data['type'] in ['Person', 'Character']:
+            artistdata['born'] = begin
+            artistdata['died'] = end
```

Comment:
Could we save "begin" and "end" from Musicbrainz to "Born" and "Died" as a catch all rather than only when type is Person of Charater, covering when no type or some new type Musicbrainz start using, rather than lose data

Reply:
currently, only the type="other" is missing. 
this type is used for things like a collection of nature recordings (for instance):
https://musicbrainz.org/ws/2/artist/51118c9d-965d-4f9f-89a1-0091837ccf54?inc=url-rels+release-groups&type=album&fmt=json

whenever musicbrainz updates their api (last time was 2017, next update is scheduled for may, 13th) we should review the changes and update the scrapers accordingly (if needed).